src/java/main/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -310,8 +311,8 @@ public boolean initiateConnection(Socket sock, Long sid) {
      * possible long value to lose the challenge.
      *
      */
-    public void receiveConnection(Socket sock) {
-        Long sid = null, protocolVersion = null;
+    public void receiveConnection(Socket sock) throws ConfigException {
+        Long sid, protocolVersion;
```

Comment:
Yes, integers whose values are between -127 and 128 (inclusive) are cached inside Integer object.

Plus, boxing/unboxing in a tight loop (not the case here!) can be a huge performance pitfall. For example:

```
for (Integer it = 0; it < 10000; ){
       it += 1;
}
```
Used to have abysmal performance numbers due to many auto boxing/unboxing .